src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs

```diff
@@ -201,11 +201,21 @@ public static void ExceptionTests()
         }
 
         // Synapse: 110003;Invalid user or password
-        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsNotAzureServer))]
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
         public static void VariousExceptionTests()
         {
-            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString);
-
+            SqlConnectionStringBuilder builder;
+            if (DataTestUtility.IsNotAzureServer())
+            {
+                builder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString);
+            }
+            else
+            {
+                // Strip the password in connection string as Authentication=Active Directory Managed Identity can not be used with a Password
+                string[] removeKeys = { "Password", "PWD" };
+                string connStr = DataTestUtility.RemoveKeysInConnStr(DataTestUtility.TCPConnectionString, removeKeys);
+                builder = new SqlConnectionStringBuilder(connStr);
+            }
```

Comment:
This will strip all Azure SQL Servers regardless of the authentication type. You should verify the authentication type if it's managed identity before taking out the user and pass.

Reply:
Added checking first if connecting to Azure Server and using Managed Identity authentication before stripping the password.